aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -678,17 +678,22 @@ Tensor & _index_put_impl_(Tensor & self, const torch::List<c10::optional<Tensor>
       "Please clone() the tensor before performing this operation. "
       "This also applies to advanced indexing e.g. tensor[indices] = tensor");
   }
+  auto value_ = value;
+  if (value.scalar_type() != self.scalar_type()) {
+    value_ = value.to(self.scalar_type());
+  }
```

Comment:
Are you going to make it fully compatible with numpy? If so, need to handle complex type as well?

Reply:
Yes, it will be better to align with numpy's behavior.